src/mag_toolkit/calibration/CalibrationLayer.py

```diff
@@ -24,25 +27,59 @@ def _write_to_csv(self, filepath: Path, createDirectory=False):
-        with pycdf.CDF(str(filepath), OFFSET_SKELETON_CDF) as offset_cdf:
-            offset_cdf["epoch"] = [value.time for value in self.values]
-            offset_cdf["offsets"][...] = np.nan_to_num(
-                [cal_value.value for cal_value in self.values], nan=CDF_FLOAT_FILLVAL
-            )
-            offset_cdf["timedeltas"] = [
-                cal_value.timedelta for cal_value in self.values
-            ]
-            offset_cdf["quality_flag"] = [
-                cal_value.quality_flag for cal_value in self.values
-            ]
-            offset_cdf["quality_bitmask"] = [
-                cal_value.quality_bitmask for cal_value in self.values
-            ]
-            offset_cdf["valid_start_datetime"] = self.validity.start
-            offset_cdf["valid_end_datetime"] = self.validity.end
-
-            offset_cdf.attrs["Generation_date"] = datetime.now()
-            offset_cdf.attrs["Data_version"] = self.version
+        OFFSET_SKELETON_CDF = (
+            "resource/imap_mag_l2-norm-offsets_20250421_20250421_v001.cdf"
+        )
+        skeleton_cdf = cdf_to_xarray(str(OFFSET_SKELETON_CDF), to_datetime=False)
+        epoch_values = [value.time for value in self.values]
+
+        offsets_values = np.nan_to_num(
+            [cal_value.value for cal_value in self.values],
+            nan=CDF_FLOAT_FILLVAL,
+        )
+
+        epoch_data = xr.Variable(
+            dims=["epoch"],
+            data=epoch_values,
+            attrs=skeleton_cdf["epoch"].attrs,
+        )
+        offsets_data = xr.Variable(
+            dims=["epoch", "axis"],
+            data=offsets_values,
+            attrs=skeleton_cdf["offsets"].attrs,
+        )
+        timedelta_var = xr.Variable(
+            dims=["epoch"],
+            data=[cal_value.timedelta for cal_value in self.values],
+            attrs=skeleton_cdf["timedeltas"].attrs,
+        )
+        qf_var = xr.Variable(
+            dims=["epoch"],
+            data=[cal_value.quality_flag for cal_value in self.values],
+            attrs=skeleton_cdf["quality_flag"].attrs,
+        )
+        qb_var = xr.Variable(
+            dims=["epoch"],
+            data=[cal_value.quality_bitmask for cal_value in self.values],
+            attrs=skeleton_cdf["quality_bitmask"].attrs,
+        )
+        offsets_dataset = xr.Dataset(
+            data_vars={
+                "epoch": epoch_data,
+                "offsets": offsets_data,
+                "timedeltas": timedelta_var,
+                "quality_flag": qf_var,
+                "quality_bitmask": qb_var,
+                "valid_start_datetime": self.validity.start,
+                "valid_end_datetime": self.validity.end,
+            },
+            coords={"axis": ["x", "y", "z"]},
+            attrs=skeleton_cdf.attrs,
+        )
+
+        offsets_dataset.attrs["Generation_date"] = str(np.datetime64("now"))
+        offsets_dataset.attrs["Data_version"] = self.version
+
```

Comment:
A bunch of magic strings in here that should be a constants so we dont have to change them in many different places and enforces compile time tooling on their references